dev-corner/README.md

```diff
@@ -0,0 +1,334 @@
+
+During this event processing, ESF will:
+- Identify the trigger `cloudwatch-logs` for the event received. We then see an error for `No input defined` and the message is placed in the replay queue.
+- Lambda shuts down.
+
+3. My replay queue now has one message, in which the `body` is:
+```
+misconfigured id
+```
+
+This `body` is the exact message that my log stream in the cloudwatch logs group sent. Since the message did not go through any shippers, there was no transformation.
+
+
+Now from this step, I will study some patches:
+
+#### Patch 1: replay queue is the only input
+
+My `config.yaml` looks like this:
+
+```yaml
+inputs:
+  - id: "arn:aws:sqs:eu-west-2:627286350134:constanca-test-esf-replay-queue"
+    outputs:
+      - args:
+          api_key: "<API key>"
+          elasticsearch_url: "<URL>"
+          es_datastream_name: "logs-esf.replay_sqs-default"
+        type: "elasticsearch"
+    type: "sqs"
+```
+
+Since the `body` was not transformed, this trigger will be the exact same one that the user defined. That is, `sqs`.
+
+The messages from the replay queue will be processed, and they will still fail since the input for the cloudwatch logs group is not defined. After the maximum amount of attempts, the message will be placed in the DLQ.
+
+
+#### Patch 2: replay queue and cloudwatch logs are inputs
+
+My `config.yaml` looks like this:
+
+```yaml
+"inputs":
+- "id": "arn:aws:sqs:eu-west-2:627286350134:constanca-test-esf-replay-queue"
+  "outputs":
+  - "args":
+      "api_key": "<Correct API Key>"
+      "elasticsearch_url": "<URL>"
+      "es_datastream_name": "logs-esf.replay_sqs-default"
+    "type": "elasticsearch"
+  "type": "sqs"
+- "id": "arn:aws:logs:eu-west-2:627286350134:log-group:constanca-test-local-esf:*"
+  "outputs":
+  - "args":
+      "api_key": "<Correct API key>"
+      "elasticsearch_url": "<URL>"
+      "es_datastream_name": "logs-esf.cloudwatch-default"
+    "type": "elasticsearch"
+  "type": "cloudwatch-logs"
+```
+
+This will fail. ESF will get the `config.yaml` file that is part of the message attributes field in the message placed in the replay queue. This is the `config.yaml` file that has the misconfigured id.
+
+
+### Conclusion on errors and replay queue
+
+- Both ingestion and configuration errors are put on the replay queue
+- Ingestion errors are transformed by the shippers, while configuration errors are not
+- Ingestion errors cause the trigger `replay-sqs` to be activated
+- Configuration errors use the trigger `sqs`, set by the user
+- The `config.yaml` to consume messages affected by ingestion errors needs to have inputs for the replay queue and for the resource that produced those messages
+- The input for the replay queue to consume ingestion errors is completely discarded. The output used will not be the one from the replay queue, but the one from the input that produced the message
+- There is no way to consume messages in the replay queue put there for configuration errors
```

Comment:
Oh, that's why the config is also stored in the message, right?

We probably need to reevaluate this behavior.

Reply:
So we are losing messages in this case?

Reply:
> So we are losing messages in this case?
We are not loosing them because they are in the replay queue, and later get sent to the DLQ. But we cannot forward them to a custom output.